jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js

```diff
@@ -37,6 +38,13 @@ YUI.add('deployment-credential-add', function() {
 
     DEFAULTCLOUD: 'google',
 
+    getInitialState: function() {
+      const info = this._getInfo();
+      return {
+        authType: info && Object.keys(info.forms)[0] || ''
```

Comment:
Should this be `info && info.forms && info.forms.length && Object.keys(info.forms)[0]`?
Otherwise, if we can safely assume that if info evaluates to true than we have at least one form, please add a comment saying that and explaining why.